### ISSUE TITLE
[Refactorings] (R) Rename method is broken

### ISSUE THREAD

**Issue by jecisc** [guillep/pharo-core#17442]
When doing a rename method refactoring from the method code pane of Calypso we have an error when trying to open a presenter.

Pharo 13.

```
UndefinedObject(Object)>>error:
UndefinedObject(Object)>>errorNotIndexable
UndefinedObject(Object)>>size
SpTextInputFieldPresenter(SpAbstractTextPresenter)>>selectAll
[ selectorInput takeKeyboardFocus; selectAll ] in StMethodNameEditorPresenter>>initializeDialogWindow: in Block: [ selectorInput takeKeyboardFocus; selectAll ]
False>>ifFalse:
[ :value | value ifFalse: aBlock ] in SpBlockedDialogWindowPresenter(SpWindowPresenter)>>whenOpenedDo: in Block: [ :value | value ifFalse: aBlock ]
FullBlockClosure(BlockClosure)>>cull:
FullBlockClosure(BlockClosure)>>cull:cull:
[ :block | block cull: self value cull: oldValue ] in ObservableValueHolder>>valueChanged: in Block: [ :block | block cull: self value cull: oldValue ]
OrderedCollection>>do:
ObservableValueHolder>>valueChanged:
[ | oldValue |
	lock := true.
	oldValue := value.
	value := anObject.
	self valueChanged: oldValue ] in ObservableValueHolder>>value: in Block: [ | oldValue |...
FullBlockClosure(BlockClosure)>>ensure:
ObservableValueHolder>>value:
SpBlockedDialogWindowPresenter(SpWindowPresenter)>>windowIsOpened
SpMorphicBlockedDialogWindowAdapter>>openWithDeferredAction:
[ :anAdapter | 
		anAdapter openWithDeferredAction: [ 
			self allPresenters do: [ :each | each announceDisplayed ].
			self updateTitle ] ] in SpBlockedDialogWindowPresenter(SpWindowPresenter)>>openWithLayout: in Block: [ :anAdapter | ...
SpBlockedDialogWindowPresenter(SpAbstractPresenter)>>withAdapterDo:
SpBlockedDialogWindowPresenter(SpWindowPresenter)>>openWithLayout:
SpBlockedDialogWindowPresenter(SpWindowPresenter)>>openWith:
SpBlockedDialogWindowPresenter(SpWindowPresenter)>>open
StMethodNameEditorPresenter(SpAbstractPresenter)>>openBlockedDialog
StMethodNameEditorPresenter class>>openOn:withInvalidArgs:canRenameArgs:canRemoveArgs:canAddArgs:
[ :methodName :ref | | invalidArgs |
		invalidArgs := self computeInvalidArgNamesForSelector: method selector.
		dialog := StMethodNameEditorPresenter
			openOn: methodName
			withInvalidArgs: invalidArgs
			canRenameArgs: true
			canRemoveArgs: false
			canAddArgs: false.
		dialog cancelled ifTrue: [ CmdCommandAborted signal ].
		methodName ] in SycExtractMethodAndOccurrencesCommand>>setUpOptionToChangeMethodNameDuring: in Block: [ :methodName :ref | | invalidArgs |...
FullBlockClosure(BlockClosure)>>cull:cull:
RBExtractMethodRefactoring(ReAbstractTransformation)>>requestMethodNameFor:
RBExtractMethodRefactoring>>getNewMethodName
RBExtractMethodRefactoring>>privateTransform
RBExtractMethodRefactoring(ReAbstractTransformation)>>generateChanges
RBExtractMethodAndOccurrences(ReAbstractTransformation)>>generateChangesFor:
RBExtractMethodAndOccurrences>>extractMethod
RBExtractMethodAndOccurrences>>privateTransform
RBExtractMethodAndOccurrences(ReAbstractTransformation)>>generateChanges
[ :each | | model |
		model := (RBNamespace onEnvironment: rbEnvironment )
				name: 'Changes for ', each class name asString;
				yourself.
		each model: model.
		each generateChanges ] in SycRefactoringPreviewPresenter>>generateChanges in Block: [ :each | | model |...
Array(SequenceableCollection)>>do:
SycRefactoringPreviewPresenter>>generateChanges
SycRefactoringPreviewPresenter>>updateChanges
[ :scope | self updateChanges ] in SycRefactoringPreviewPresenter>>initializePresenter in Block: [ :scope | self updateChanges ]
FullBlockClosure(BlockClosure)>>cull:
[	aBlock cull: self selectedItem ] in SpDropListPresenter>>whenSelectedItemChangedDo: in Block: [	aBlock cull: self selectedItem ]
FullBlockClosure(BlockClosure)>>cull:
[ aBlock cull: self ] in SpSingleSelectionMode>>whenChangedDo: in Block: [ aBlock cull: self ]
FullBlockClosure(BlockClosure)>>cull:
FullBlockClosure(BlockClosure)>>cull:cull:
[ :block | block cull: self value cull: oldValue ] in ObservableValueHolder>>valueChanged: in Block: [ :block | block cull: self value cull: oldValue ]
OrderedCollection>>do:
ObservableValueHolder>>valueChanged:
[ | oldValue |
	lock := true.
	oldValue := value.
	value := anObject.
	self valueChanged: oldValue ] in ObservableValueHolder>>value: in Block: [ | oldValue |...
FullBlockClosure(BlockClosure)>>ensure:
ObservableValueHolder>>value:
SpSingleSelectionMode>>basicSelectIndex:
SpSingleSelectionMode(SpAbstractSelectionMode)>>selectIndex:
SpDropListPresenter>>selectIndex:
SpDropListPresenter>>setCollection:
SpDropListPresenter>>items:
SycRefactoringPreviewPresenter>>scopes:
SycRefactoringPreviewPresenter class>>for:scopes:
SycExtractMethodAndOccurrencesCommand(CmdCommand)>>confirmRefactoringInContext:by:
ClyMethodSourceCodeContext(ClySourceCodeContext)>>executeCommand:by:
[self prepareCommandForExecution.
	context executeCommand: command by: self.
	self applyCommandResult] in CmdCommandActivator>>executeCommand in Block: [self prepareCommandForExecution....
FullBlockClosure(BlockClosure)>>on:do:
CmdCommandActivator>>executeCommand
CmdCommandMenuItem>>executeCommand
[aCommandItem executeCommand] in SycExtractMethodAndOccurrencesCommand(CmdCommand)>>doRegisterContextMenuItemsFor:withBuilder: in Block: [aCommandItem executeCommand]
```

**Comment by Ducasse** [comment]
I could not reproduce it. I tried 

- Selecting the selector.
- Placing the caret inside a message in the method body
- in the method list 

Now I got a strange pop when I selected a message in the method body

![Capture 2024-11-25 at 20 33 20](https://github.com/user-attachments/assets/d939d9a1-2225-4eb6-880b-65dba6f88da7)

**Comment by Ducasse** [comment]
Obviously it does not define it because I'm rename the method to this new name.

**Comment by jecisc** [comment]
Hum... maybe this was Extract Method and not Rename Method.
I might have mixed both I. My head

**Comment by Ducasse** [comment]
Ok to be investigated.

**Comment by AlexisCnockaert** [comment]
Ok there were several bugs. 
The extract method did not replace the extracted nodes by the extracted method selector. 
This is due to these changes: https://github.com/pharo-project/pharo/pull/17001/files#diff-7a5d1f94e8e09953bdd828ffbf4f004f12311cd6eff8dfad8b13bee3ed204249

We did a dichotomy checks and reverting the second changes fixes the problem.
We believe the problem is that the model are not shared anymore. 

<blockquote><img src="https://avatars.githubusercontent.com/u/4822303?s=400&v=4" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/pharo-project/pharo/pull/17001">Fix Issue #16945: Refactorings -> add argument: fails on unary messages by carolahp · Pull Request #17001 · pharo-project/pharo</a></strong></div><div>Fix Issue#16945
Replaces PR#16951
When the method generateChanges is sent, it checks for preconditions and then it performs the corresponding changes on the object rb-class (which represents the re...</div></blockquote>
